packages/data-stores/src/wpcom-plans-ui/reducer.ts

```diff
@@ -14,8 +15,19 @@ const showDomainUpsellDialog: Reducer< boolean | undefined, WpcomPlansUIAction >
 	return state;
 };
 
+const selectedStorageAddOnsForPlans: Reducer<
+	selectedStorageAddOnsForPlans | undefined,
+	WpcomPlansUIAction
+> = ( state, action ) => {
+	if ( action.type === 'WPCOM_PLANS_UI_SET_STORAGE_ADD_ON_FOR_PLAN' ) {
+		return { ...state, [ action.plan ]: action.addOnSlug };
+	}
+	return state;
+};
+
 const reducer = combineReducers( {
 	showDomainUpsellDialog,
+	selectedStorageAddOnsForPlans,
```

Comment:
Asking here, so we can resolve in a thread (so not exactly related to this piece of code).

Do we expect to retrieve the selected storage from a location outside of the plans grid package (`plan-features-2023-grid`)? e.g. from `checkout` maybe? If that's the case, then we will need to set the value to `persist` in the data store, and possibly call an action to reset it when rendering the grid components (features table or comparison grid) in a "fire once" hook.

Reply:
> Do we expect to retrieve the selected storage from a location outside of the plans grid package (plan-features-2023-grid)? e.g. from checkout maybe?

Not for the time being. `checkout` already has their own state management in place for add ons